cli/tests/test_diff_command.rs

```diff
@@ -2274,6 +2275,59 @@ fn test_diff_external_tool() {
     ");
 }
 
+#[test]
+fn test_diff_do_chdir() {
+    let test_env = TestEnvironment::default();
+    test_env.run_jj_in(".", ["git", "init", "repo"]).success();
+    let work_dir = test_env.work_dir("repo");
+    work_dir.write_file("file1", "file1\n");
+
+    assert_eq!(
+        work_dir
+            .run_jj([
+                "diff",
+                "--tool=pwd",
+                "--config=merge-tools.pwd.diff-args=[]"
+            ])
+            .to_string()
+            .lines()
+            .next()
+            .unwrap(),
+        "$TEST_ENV/repo" // I don't know why pwd returns this, but it does
+    );
+    assert!(
+        work_dir
+            .run_jj([
+                "diff",
+                "--tool=pwd",
```

Comment:
To have the command work on Windows, you might need to use something like https://github.com/jj-vcs/jj/blob/main/cli/testing/fake-diff-editor.rs and add a command there.

I suppose you could also use Windows' analogue of `pwd`, but that seems like more of a headache.

Reply:
Actually, you might not need to add any commands, as long as `fake-diff-editor` can print the file paths and check that it can read the files.

This might also avoid the need for messing with `TMP_DIR`, https://github.com/jj-vcs/jj/pull/6184#discussion_r2023795109